sharedbuffers/mapped_struct.py

```diff
@@ -885,7 +941,14 @@ def _eq(self, other):
             return proxied_list_cmp(self, other) == 0
 
     def __len__(self):
-        return self._metadata()[1]
+        if self.elem_step == 0:
+            return self._metadata()[1]
+        elif self.elem_start == self.elem_end:
```

Comment:
shouldn't it return 0 if `self.elem_start <= self.elem_end`?

Reply:
Nah, those formulas work in most cases, _except_ when given malicious input, hence the need to manually handle them.